lib/pure/httpclient.nim

```diff
@@ -997,12 +997,14 @@ proc requestAux(client: HttpClient | AsyncHttpClient, url: Uri,
 
   await newConnection(client, url)
 
-  let newHeaders = client.headers.override(headers)
+  var newHeaders: HttpHeaders
 
   var data: seq[string]
   if multipart != nil and multipart.content.len > 0:
     data = await client.format(multipart)
+    newHeaders = client.headers.override(headers)
```

Comment:
can you explain (as comment in the code) why `let newHeaders = client.headers.override(headers)` would not work (instead of calling it in each branch), so that this doesn't regress in future

Reply:
First maybe explain here, I see the change as not producing any semantic changes, am I missing something?

Reply:
Hi, firstly thank you all for the direction (this is my first Nim PR, so every bit of info is helpful).  
Secondly, the issue is that when there is multipart data, a side effect of calling `client.format(multipart)` is that the content-length and content-type headers on the client are set to correct values.  The way that `proc requestAux `was structured meant that the client headers were copied to `newHeaders` before `client.format(mulipart)` was called, and then `newHeaders` was used to generate the request.  This meant that the changes to the headers from `client.format(multipart)` were lost.

The solution I chose is to populate `newHeaders` after `client.format(multipart)` only when there is multipart data. When there is no multipart data, `newHeaders` is populated before the Content-Length value is assessed on that branch so that the caller doesn't loose the the option of overriding the Content-Length header with a custom value when needed.

Reply:
ya i suspected something like that; you could add a 1 sentence comment along the lines of `format resets headers` and link to https://github.com/nim-lang/Nim/pull/18208#discussion_r647036979

Reply:
Will do.   It'll take a little time to get back to the PR due to my current workload, but it won't get forgotten.